### PR TITLE
Added precursor teleporer sync

### PR DESCRIPTION
**Pull request by Coding-Hen**
Sync currently isnt persistant as with other "doors"

### CODE REVIEW

File: NitroxClient/GameLogic/Spawning/Metadata/PrecursorTeleporterMetadataProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+using NitroxModel.Logger;
+using UnityEngine;
+
+namespace NitroxClient.GameLogic.Spawning.Metadata
+{
+    public class PrecursorTeleporterMetadataProcessor : GenericEntityMetadataProcessor<PrecursorTeleporterMetadata>
+    {
+        public override void ProcessMetadata(GameObject gameObject, PrecursorTeleporterMetadata metadata)
+        {
+            Log.Info($"Received precursor teleporter metadata change for {gameObject.name} with data of {metadata}");
```
Comment (Jannify) [inline NitroxClient/GameLogic/Spawning/Metadata/PrecursorTeleporterMetadataProcessor.cs:11]: Maybe Log.Debug? The `PrecursorTeleporterMetadata`-parameter seems more like debug but idk....

File: NitroxClient/GameLogic/Spawning/Metadata/PrecursorTeleporterMetadataProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+            PrecursorTeleporter precursorTeleporter = gameObject.GetComponent<PrecursorTeleporter>();
```
Comment (Jannify) [inline NitroxClient/GameLogic/Spawning/Metadata/PrecursorTeleporterMetadataProcessor.cs:13]: Maybe check if the component is found, just to be save.

File: NitroxClient/GameLogic/Spawning/Metadata/PrecursorTeleporterMetadataProcessor.cs (hunk 1 above)
Comment (Coding-Hen) [reply NitroxClient/GameLogic/Spawning/Metadata/PrecursorTeleporterMetadataProcessor.cs:11]: This change can be made but its simular to the KeypadMetadataProcessor sunrunner created, so was just following that convention

**Comment by Sunrunner37** [comment]
I think this is same behavior as keypad is to starship door? let's align on that one first.

**Pull request merged**